mesonbuild/backend/backends.py

```diff
@@ -748,6 +748,7 @@ def get_external_rpath_dirs(self, target: build.BuildTarget) -> T.Set[str]:
                         raise MesonException(f'Invalid arg for --just-symbols, {dir} is a directory.')
         return dirs
 
+    @lru_cache(maxsize=None)
     def rpaths_for_bundled_shared_libraries(self, target: build.BuildTarget, exclude_system: bool = True) -> T.List[str]:
```

Comment:
Can you change the return type of this to `ImmutableListProto[str]`? This tells the static type checker that the return of this shouldn't be mutated

Reply:
Basically the same thing done by the `get_custom_target_provided_by_generated_source` and `get_custom_target_provided_libraries` functions.

Note: it is ImmutableListProtocol with a "col" at the end. :p